src/components/flow/nodes/APINode.tsx

```
@@ -0,0 +1,115 @@
+		return <div className="text-red-500">No node data found</div>;
+	}
+
+	// Type assertion after the type check
+	const nodeidl = nodeIdls.get(props.data.data.nodeidl);
+
+	if (!nodeidl) {
+		return <div className="text-red-500">Invalid nodeidl: {props.data.data.nodeidl}</div>;
+	}
+
+	const inpValues = props.data.data.inputValues;
+	const inp = useMemo(() => {
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

Hooks should not be called after an early return.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)